### PR TITLE
sidecar: change restore.sh to go codes #292

### LINKED ISSUE #292: [feature] change restore.sh to go codes
**Is your feature request related to a problem? Please describe.**
remove the restore.sh script, build it by using go codes

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by acekingke**
### What type of PR is this?

/enhancement
### Which issue(s) this PR fixes?

Fixes #292 

### What this PR does?

Summary:

### CODE REVIEW

File: sidecar/config.go [hunk 1]
```diff
@@ -552,53 +551,55 @@ func (cfg *Config) buildS3Restore(path string) error {
 		len(cfg.XCloudS3Bucket) == 0 {
 		return fmt.Errorf("do not have S3 information")
 	}
-	f, err := os.Create(path)
-	if err != nil {
-		return fmt.Errorf("create restore.sh fail : %s", err)
+	//check has directory, and create it.
```
Comment (runkecheng) [inline sidecar/config.go:554]: fix note. like // Check has directory, and create it.

**Review by runkecheng — CHANGES_REQUESTED** [review]

File: sidecar/config.go [hunk 2]
```diff
@@ -552,53 +551,77 @@ func (cfg *Config) buildS3Restore(path string) error {
 		len(cfg.XCloudS3Bucket) == 0 {
 		return fmt.Errorf("do not have S3 information")
 	}
-	f, err := os.Create(path)
-	if err != nil {
-		return fmt.Errorf("create restore.sh fail : %s", err)
+	// Check has directory, and create it.
+	if _, err := os.Stat(utils.DataVolumeMountPath); os.IsNotExist(err) {
+		if err := os.MkdirAll(utils.DataVolumeMountPath, 0755); err != nil {
+			return fmt.Errorf("create data directory fail : %s", err)
```
Comment (runkecheng) [inline sidecar/config.go:557]: `failed to xxx` is better?

File: sidecar/config.go [hunk 3]
```diff
@@ -552,53 +551,77 @@ func (cfg *Config) buildS3Restore(path string) error {
...
+		}
+	}
+	// mkdir /root/backup.
+	if err := os.MkdirAll("/root/backup", 0755); err != nil {
+		return fmt.Errorf("create backup directory fail : %s", err)
```
Comment (runkecheng) [inline sidecar/config.go:562]: same as the last.

**Review by runkecheng — CHANGES_REQUESTED** [review]

**Review by runkecheng — APPROVED** [review]

**Comment by runkecheng** [comment]
lgtm

File: sidecar/config.go [hunk 4]
```diff
@@ -554,53 +553,77 @@ func (cfg *Config) buildS3Restore(path string) error {
...
---parallel=10 {{.XRestoreFrom}} \
---insecure |xbstream -xv -C /root/backup
-# prepare redolog
-xtrabackup --defaults-file={{.MyCnfMountPath}} --use-memory=3072M --prepare --apply-log-only --target-dir=/root/backup
-# prepare data
-xtrabackup --defaults-file={{.MyCnfMountPath}} --use-memory=3072M --prepare --target-dir=/root/backup
-chown -R mysql.mysql /root/backup
-xtrabackup --defaults-file={{.MyCnfMountPath}} --datadir={{.DataDir}} --copy-back --target-dir=/root/backup
-chown -R mysql.mysql {{.DataDir}}
-rm -rf /root/backup
```
Comment (andyli029) [inline sidecar/config.go:586]: The step is ignored.

**Review by andyli029 — COMMENTED** [review]
LGTM.

**Pull request merged**